enterprise/backend/src/metabase_enterprise/semantic_search/indexer.clj

```diff
@@ -92,9 +222,16 @@
 (defn on-indexing-idle
   "Runs any loop idle behaviour (such as waiting), called after each step."
   [indexing-state]
-  (let [{:keys [last-poll-count last-indexed-count]} @indexing-state
-        novelty-ratio (if (zero? last-poll-count) 0 (/ last-indexed-count last-poll-count))]
+  (let [{:keys [last-poll-count
+                last-novel-count
+                ^Instant next-dlq-run]} @indexing-state
+        novelty-ratio (if (zero? last-poll-count) 0 (/ last-novel-count last-poll-count))]
+
     (cond
+      ;; if the next DLQ run should happen on the next iteration, we should not idle
+      (and next-dlq-run (.isAfter (.instant clock) next-dlq-run))
```

Comment:
order was swapped here: ensure we *do* idle unless `next-dlq-run` is ready

https://github.com/metabase/metabase/pull/62070/commits/69839a5d210567e52ac96657440c84506abda8be